src/System.IO.Pipelines/src/System/IO/Pipelines/ThreadPoolScheduler.NetCoreApp21.cs

```diff
@@ -9,16 +9,9 @@ namespace System.IO.Pipelines
 {
     internal sealed class ThreadPoolScheduler : PipeScheduler
     {
-        public override void Schedule(Action action)
-        {
-            System.Threading.ThreadPool.QueueUserWorkItem(s_actionCallback, action, preferLocal: false);
-        }
-
-        public override void Schedule(Action<object> action, object state)
+        public override void Schedule<TState>(Action<TState> action, TState state)
         {
             System.Threading.ThreadPool.QueueUserWorkItem(action, state, preferLocal: false);
```

Comment:
@stephentoub Question for you, since we're doing all of the capture and restore in the pipe implementations I was thinking we should use UnsafeQueueUserWorkItem here for a default fast path. There's no point saving and restoring the execution context here if it's being done several layers about these calls. (BTW that would mean we should also consider adding `UnsafeQueueUserWorkItem(Action<object>, object, preferLocal)`

Alternatively I though about copying the "Unsafe" moniker here to denote that there can be a code path that doesn't capture the ExecutionContext. 

Any thoughts here?

Reply:
>  I was thinking we should use UnsafeQueueUserWorkItem here for a default fast path.

If the only way to get here would have already done the right thing with regards to capturing if needed and ensuring it's subsequently restored, that's a fine young to do.

> BTW that would mean we should also consider adding UnsafeQueueUserWorkItem(Action<object>, object, preferLocal)

Yeah, I was very close to just doing that when I fixed up the existing overload.